CHANGES.rst

```diff
@@ -141,9 +141,12 @@ tweakreg
 
 - Update ``sregion`` after WCS corrections are applied. [#8158]
 
-- Added option to choose IRAFStarFinder and segmentation.SourceFinder
+- Add option to choose IRAFStarFinder and segmentation.SourceFinder
   instead of DAOStarFinder and exposed star finder parameters. [#8203]
 
+- Suppress warnings from ``photutils.background.Background2D`` regarding
```

Comment:
I have not seen `SourceCatalogStep` spit out this warning.  Probably that is because the input data, resampled _i2d.fits files, does not have NaNs in it.  But I thought it best to be consistent in the filtering, just in case.

Funny enough, in `SourceCatalogStep`, the NaNs are explicitly masked in the `mask` which is then passed to `Background2D`, but this still doesn't seem to alleviate the warning from `photutils`.  I.e. if I have an image with NaNs, pass it to `Background2D` with a mask that masks the NaNs, I still get the warning.  I'll see if I can fix that in `photutils`.

I will add the CHANGES.rst line as well.

Reply:
See astropy/photutils#1711

Reply:
But my reason for saying you should add an entry under `source_catalog` in the change log is because this PR contains changes to `source_catalog/detection.py` to suppress warnings, just like the updates to `tweakreg`.

Reply:
Done.